src/agent/communicator/include/communicator.hpp

```diff
@@ -1,9 +1,12 @@
 #pragma once
 
 #include "defs.hpp"
+#include <../../configuration_parser/include/i_configuration_parser.hpp>
```

Comment:
We shouldn't need to include a header using a relative path to a parent folder... maybe we'll have to find an alternate solution.

Btw, since the `iConfigurationParser` is used in this file only as a pointer, we can forward declare it, and put the include on the source file.

Reply:
Solved.